client/import_certify_test.go

```diff
@@ -0,0 +1,257 @@
+package client_test
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"strings"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/internal/test"
+	tpb "github.com/google/go-tpm-tools/proto/tpm"
+	"github.com/google/go-tpm-tools/server"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpm2/transport"
+)
+
+func TestCreateCertifiedAKBlob(t *testing.T) {
+	rwc := test.GetTPM(t)
+	tpm := transport.FromReadWriter(rwc)
+	defer client.CheckedClose(t, rwc)
+
+	ek, err := client.EndorsementKeyRSA(rwc)
+	if err != nil {
+		t.Fatalf("failed to get rsa ek: %v", ek)
+	}
+	defer ek.Close()
+
+	pubBytes, err := ek.PublicArea().Encode()
+	if err != nil {
+		t.Fatalf("ek public area encode failed: %v", err)
+	}
+	pub, err := tpm2.Unmarshal[tpm2.TPMTPublic](pubBytes)
+	if err != nil {
+		t.Fatalf("Unmarshal public key failed: %v", err)
+	}
+
+	testcases := []struct {
+		name    string
+		keyAlgo tpm2.TPMAlgID
+	}{
+		{"RSA", tpm2.TPMAlgRSA},
+		{"ECC", tpm2.TPMAlgECC},
+	}
+	for _, tc := range testcases {
```

Comment:
nit: could the CreateCertifiedAKBlob be a receiver method on the ek Key. You can then call ek.PublicArea() below and ek.CreateCertifiedAKBlob(tpm, challenge, tc.keyAlgo)